### DISCUSSION TITLE
Heat sensor - for air temp

### DISCUSSION THREAD

**Discussion by kgreene1983** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#280] [Ideas]
Is it possible to add a heat sensor so that you can tell the outdoor temperature (well the temp inside the egg).

**Comment by visualapproach** [comment]
Haven't tested but should be doable by connecting to d0 or d8

**Reply by visualapproach** [reply]
And adding some code to handle it ofc. I won't add that to the project but I can support if you need help. One thing to bear in mind:
<html>
<body>


Label | GPIO | Input | Output | Notes
-- | -- | -- | -- | --
D0 | GPIO16 | no interrupt | no PWM or I2C support | **HIGH at boot used to wake up from deep sleep**
D1 | GPIO5 | OK | OK | often used as SCL (I2C)
D2 | GPIO4 | OK | OK | often used as SDA (I2C)
D3 | GPIO0 | pulled up | OK | connected to FLASH button, boot fails if pulled LOW
D4 | GPIO2 | pulled up | OK | HIGH at bootconnected to on-board LED, boot fails if pulled LOW
D5 | GPIO14 | OK | OK | SPI (SCLK)
D6 | GPIO12 | OK | OK | SPI (MISO)
D7 | GPIO13 | OK | OK | SPI (MOSI)
D8 | GPIO15 | pulled to GND | OK | SPI (CS)Boot fails if pulled HIGH
RX | GPIO3 | OK | RX pin | HIGH at boot
TX | GPIO1 | TX pin | OK | HIGH at bootdebug output at boot, **boot fails if pulled LOW**
A0 | ADC0 | Analog Input | X | 


</body>
</html>

**Reply by kgreene1983** [reply]
Can I use a DALLAS DS18B20 18B20 TO-92 IC CHIP Thermometer Temperature Sensor? 

It would be cool to see the outside temperature.

**Reply by visualapproach** [reply]
I tried @ pins D0, D8 and A0 with no luck. The only way I can think of is to sacrifice the audio pin. Connect the audio pins between CIO and DSP - you will have the standard sound. And use pin D6 (disconnected from the DSP/CIO) for the Dallas chip. Then it may work. The library for DS18B20 turns off interrupts periodically which may interfere with the layzspa library. You can use 3.3V directly to this chip.
